examples/subscriber_simple.rs

```diff
@@ -2,38 +2,25 @@
 // SPDX-FileCopyrightText: © Contributors to the iceoryx-rs project
 // SPDX-FileContributor: Mathias Kraus
 
-use iceoryx_rs::sb::{SubscribeState, TopicBuilder};
+use iceoryx_rs::sb::SubscriberBuilder;
 use iceoryx_rs::Runtime;
 
+use std::error::Error;
 use std::thread;
 use std::time::Duration;
 
 #[repr(C)]
-struct CounterTopic {
+struct Counter {
```

Comment:
Would it be possible to outsource this into a separate file so that every publisher and subscriber share the same topic definition.
Otherwise it can become easily inconsistent.

Reply:
Will do this in a follow up PR. I need to refactor the example a bit in order to share files between the examples. While it is dead easy to add simple examples with cargo, by default it expects one file per example. It is possible to add more complex example but it is ... well ... also a bit more complex